File: PalindromeNumber.java
public class Solution {
	public boolean isPalindrome(int x) {
		if(x < 0)
			return false;
		int temp = x;
		int check = 0;
		while(x > 0) {
			check = check * 10;
			check += x % 10;
			x /= 10;
		}
		return check == temp;
	}
}

File: PalindromeNumber.cc
class Solution {
    public:
        bool isPalindrome(int x) {
            if (x < 0) {
                return false;
            }

            string str_x = to_string(x);
            /*if (str_x.size() % 2 == 1) {
             *             return false;
             *                     }
             *                             */
            int index = 0;
            while (index < str_x.size() / 2) {
                char current = str_x[index];
                if (current != str_x[str_x.size() - 1 - index]) {
                    return false;
                } else {
                    index++;
                }
            }

            return true;
        }
};
